File: modules/wem-api/src/main/java/com/enonic/wem/api/content/page/PageTemplateName.java
package com.enonic.wem.api.content.page;

import com.enonic.wem.api.content.ContentName;

public class PageTemplateName
    extends ContentName
{
    public PageTemplateName( final String name )
    {
        super( name );
    }

    @Override
    public boolean equals( final Object o )
    {
        if ( this == o )
        {
            return true;
        }
        if ( o == null || getClass() != o.getClass() )
        {
            return false;
        }

        final PageTemplateName that = (PageTemplateName) o;
        return super.equals( that );
    }

    public static PageTemplateName from( final String name )
    {
        return new PageTemplateName( name );
    }

    public static PageTemplateName from( final ContentName name )
    {
        return new PageTemplateName( name.toString() );
    }
}

File: modules/admin/admin-ui/src/main/resources/web/admin/common/js/content/page/PageTemplateName.ts
module api.content.page{

    export class PageTemplateName {

        private name:string;

        constructor(name:string) {
            if( name == null ) {
                throw new Error("name cannot be null");
            }
            this.name = name;
        }

        public toString():string {
            return this.name;
        }
    }
}